components/style/gecko/wrapper.rs

```diff
@@ -1136,6 +1146,68 @@ impl<'le> ::selectors::Element for GeckoElement<'le> {
         None
     }
 
+    fn attr_matches(&self,
+                    ns: &NamespaceConstraint<&Namespace>,
+                    local_name: &Atom,
+                    operation: &AttrSelectorOperation<&Atom>)
+                    -> bool {
+        unsafe {
+            match *operation {
+                AttrSelectorOperation::Exists => {
```

Comment:
it'd be nice to share this huge switch with the snapshot code, perhaps it not too terrible to do this with something on the line of `snapshot_helpers`?